src/monitor.ts

```diff
@@ -160,27 +160,37 @@ export class Monitor {
   /**
    * Start the embedded HTTP server to process requests on the given port.
    */
-  public start(port: number) {
+  public async start(port: number): Promise<void> {
     this.timer = this.createPingTimer();
     this.ping();
 
-    this.server = http.createServer(this.app);
-    this.server.listen(port);
+    const promise = new Promise<void>(resolve => {
```

Comment:
just return here? (same below)

Reply:
I want the promise to only resolve after `listen()` has finished, which occurs in the callback, would returning here have that behavior?

Reply:
afaik yes. assigning it to a variable shouldn't change anything.

Reply:
oh oh I misunderstood. I will fix this